src/clj/rems/db/entitlements.clj

```diff
@@ -63,29 +63,42 @@
        :mail (:mail e)
        :end (:end e)})))
 
-(defn- post-entitlements! [{:keys [entitlements action] :as params}]
-  (when-let [target (get-in env [:entitlements-target action])]
-    (let [payload (get-entitlements-payload entitlements action)
-          json-payload (json/generate-string payload)]
-      (log/infof "Posting entitlements to %s:" target payload)
-      (let [response (try
-                       (http/post target
-                                  {:throw-exceptions false
-                                   :body json-payload
-                                   :content-type :json
-                                   :socket-timeout 2500
-                                   :conn-timeout 2500})
-                       (catch Exception e
-                         (log/error "POST failed" e)
-                         {:status "exception"}))
-            status (:status response)]
-        (when-not (= 200 status)
-          (log/warnf "Entitlement post failed: %s", response)
-          (str "failed: " status))))))
+(defn- post-entitlements! [{:keys [action type entitlements config]}]
+  (case type
+    :ega (when config
+           (try (doseq [entitlement entitlements] ; technically these could be grouped per user & api-key
+                  (ega/entitlement-push action entitlement config))
+                (catch Exception e
+                  (log/error "POST failed" e)
+                  (or (ex-data e)
+                      {:status "exception"}))))
+
+    :basic ; TODO: let's move this :entitlements-target (v1) at some point to :entitlement-post (v2)
+    (when-let [target (get-in env [:entitlements-target action])]
+      (let [payload (get-entitlements-payload entitlements action)
+            json-payload (json/generate-string payload)]
+        (log/infof "Posting entitlements to %s: %s" target payload)
+        (let [response (try
+                         (http/post target
+                                    {:throw-exceptions false
+                                     :body json-payload
+                                     :content-type :json
+                                     :socket-timeout 2500
+                                     :conn-timeout 2500})
+                         (catch Exception e
+                           (log/error "POST failed" e)
+                           {:status "exception"}))
+              status (:status response)]
+          (when-not (= 200 status)
+            (log/warnf "Entitlement post failed: %s", response)
+            (str "failed: " status)))))))
 
 ;; TODO argh adding these everywhere sucks
 (defn- fix-entry-from-db [entry]
-  (update-in entry [:outbox/entitlement-post :action] keyword))
+  (-> entry
+      (update-in [:outbox/entitlement-post :action] keyword)
+      (update-in [:outbox/entitlement-post :type] keyword)
+      (update-in [:outbox/entitlement-post :config :type] keyword)))
```

Comment:
could consider a schema for these coersions?

Reply:
I shall add a TODO about it here.